aten/src/ATen/test/math_kernel_test.cpp

```diff
@@ -4,9 +4,22 @@
 
 using namespace at;
 
-#define ASSERT_ALLCLOSE_TOLERANCES(t1, t2, atol, rtol) \
-  ASSERT_TRUE(t1.is_same_size(t2));                    \
-  ASSERT_TRUE(t1.allclose(t2, atol, rtol));
+bool allClose(const at::Tensor& t1, const at::Tensor& t2, double rtol=1e-5, double atol=1e-8) {
+  if (!t1.is_same_size(t2)) {
+    std::cerr << "Difference in tensor shapes: "
+      << t1.sizes() << " v.s. " << t2.sizes() << std::endl;
+    return false;
+  }
+  bool equal = t1.allclose(t2, rtol, atol);
+  if (!equal) {
+    std::cerr << "Difference in tensor value: \nFirst tensor:\n"
+        << t1 << "\nSecond tensor:\n" << t2 << std::endl;
```

Comment:
This is cool, although it may be hard to spot where the tensors are different. Still seems like an improvement, especially when t1 and t2 are small.

Reply:
Yea I think having max difference and their location is useful in local debugging, but it's generally more useful to have the full tensor available when shown on CI(especially when it comes to setting atol/rtol). Let's try this first, if it needs more detailed inforation later I can add them too :D